packages/loader/driver-utils/src/parallelRequests.ts

```diff
@@ -474,13 +486,31 @@ export function requestOps(
             queue.pushValue(deltas);
         });
 
+    // Implement faster cancellation. getSingleOpBatch() checks signal, but only in between
+    // waits (up to 10 seconds) and fetches (can take infinite amount of time).
+    // While every such case should be improved and take into account signal (and thus cancel immediately),
+    // it is beneficial to have catch-all
+    const listener = (event: Event) => { manager.cancel(); };
+    if (signal !== undefined) {
+        signal.addEventListener("abort", listener);
+    }
+
     manager.run(concurrency)
-        .then(() => {
-            telemetryEvent.end({
+        .finally(() => {
```

Comment:
Structure wise shouldn't finally come in end.

Reply:
Yep, left behind after experimenting with adding props, will fix it.

Reply:
It does not matter, as it ensures that code runs after run() is over - the order does not matter here.